### PR TITLE
docs: make toc sticky on desktop

### PR DESCRIPTION
**Pull request by chenxsan**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update

Closes https://github.com/eslint/eslint/issues/16312

[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

1. Make the table of contents sticky on desktop:

    CleanShot 2022-11-06 at 22 33 59@2x

2. Highlight respective menu item in table of contents (light blue area is for demonstration only, not included in code):

    image

### CODE REVIEW

File: docs/src/assets/js/main.js [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+                    `a[href="#${entry.target.id}"]`
+                );
+                if (nextActiveAnchor) {
+                    nextActiveAnchor.parentNode.classList.add(activeClassName);
+                    nextActiveAnchor.classList.add(activeClassName);
+                }
+            }
+        });
+    }, options);
+    if (window.matchMedia("(min-width: 1400px)").matches) {
```
Comment (chenxsan) [inline docs/src/assets/js/main.js:31]: Only observe when table of contents in sidebar is shown.

**Comment by snitin315** [comment]
The scrollbar color is black in dark mode.
<details>
<summary>   dark mode  </summary>

https://user-images.githubusercontent.com/46647141/200472857-5aed7de9-5f07-4f9b-90ad-106d704ff2b3.mov

</details>

**Comment by chenxsan** [comment]
> The scrollbar color is black in dark mode.
> 
> dark mode

Mind sharing which browser is it?

**Comment by chenxsan** [comment]
> It's chrome v107 on MacOS

Unfortunately I can't reproduce it on my Chrome.

Can you check the horizontal scrollbar in https://eslint.org/docs/latest/user-guide/command-line-interface#options section?

Here's how it looks like on my Chrome:

**Comment by snitin315** [comment]
It looks like this
https://user-images.githubusercontent.com/46647141/200560310-983b64f0-b655-4f61-8c19-8cf1efbe1c4b.mov

**Comment by chenxsan** [comment]
> It looks like this
> 
>  Screen.Recording.2022-11-08.at.5.39.32.PM.mov

Seems we have different option regarding the scrollbar:



If I set the option to `When scrolling`, I can reproduce it then.

Anyway, seems it can be fixed by setting a background color.

**Pull request merged**